rpc_test.go

```diff
@@ -155,7 +186,55 @@ func TestMultiPingPong(t *testing.T) {
 	}
 	assert.Len(t, pongs, 2)
 
+	// test retries; peer1 requests "pong" from peer2 and peer3, but peer2 and peer3 join topic after the request
+	t4, err := p1.NewTopic(context.Background(), "topic2", true)
+	require.NoError(t, err)
+	t4.SetEventHandler(eventHandler)
+	t4.SetMessageHandler(messageHandler)
+	fin.Add(t4)
+
+	var lk sync.Mutex
+	go func() {
+		time.Sleep(time.Second) // wait until after peer1 publishes the request
+
+		t5, err := p2.NewTopic(context.Background(), "topic2", true)
+		require.NoError(t, err)
+		t5.SetEventHandler(eventHandler)
+		t5.SetMessageHandler(messageHandler)
+		lk.Lock()
+		fin.Add(t5)
+		lk.Unlock()
+
+		t6, err := p3.NewTopic(context.Background(), "topic2", true)
+		require.NoError(t, err)
+		t6.SetEventHandler(eventHandler)
+		t6.SetMessageHandler(messageHandler)
+		lk.Lock()
+		fin.Add(t6)
+		lk.Unlock()
+	}()
+	// allow enough time for peer2 join event to be propagated.
+	ctx2, cancel2 := context.WithTimeout(context.Background(), time.Second*2)
+	defer cancel2()
+	rc2, err := t4.Publish(
+		ctx2,
+		[]byte("ping"),
+		rpc.WithMultiResponse(true),
+	)
+	require.NoError(t, err)
+	var pongs2 []struct{}
+	for r := range rc2 {
+		require.NotNil(t, r)
+		require.NoError(t, r.Err)
+		assert.Equal(t, "pong", string(r.Data))
+		assert.NotEmpty(t, r.ID)
+		pongs2 = append(pongs2, struct{}{})
+	}
+	assert.True(t, len(pongs2) >= 2, "at least 2 responses should have been received")
```

Comment:
Changed this test as sometimes the JOINED event happens more than once when a new peer joins. Shouldn't be a problem I assume? retries always cause repetitive messages.

Reply:
should be okay as long as it's documented so users can be sure to ignore duplicates